### DISCUSSION TITLE
Compiler errors with E0135.

### DISCUSSION THREAD

**Discussion by Jarsergej** [libtcod/libtcod#107] [Q&A]
![image](https://user-images.githubusercontent.com/91727997/153595507-db17966f-4a1c-4e66-9ea5-b6aee69c7243.png)
when i comp code with library, it's always shows me a lot of mistakes. Why is it so? Help pls!

**Comment by HexDecimal** [comment]
The latest versions of Libtcod require C++17.  This means compiling with `/std:c++17` or later, and using Visual Studio 2017 or later, when using the Windows toolchain.  Without C++17 important templates like `std::optional` will be missing.

Update to the latest Visual Studio and then tell it to use the  `/std:c++17` or  `/std:latest` flags.
